### PR TITLE
Add lifecycle autotracking config option

### PR DESCRIPTION
**Pull request by mscwilson**
Exposes the `lifecycleAutotracking` option in `TrackerConfiguration` on mobile.

### CODE REVIEW

File: doc/02-configuration.md [hunk 1]
```diff
@@ -34,23 +34,26 @@ Setting a custom POST path can be useful in avoiding adblockers; it replaces the
...
+| `applicationContext`    | `bool?` | Indicates whether [application](http://iglucentral.com/schemas/com.snowplowanalytics.mobile/application/jsonschema/1-0-0) entity should be attached to tracked events.            | ✔ | ✔ | | true                                          |
+| `webActivityTracking`   | WebActivityTracking?` | Enables activity tracking using page views and pings on the Web.                                                                                                                  | | | ✔ | true                                          |
+| `userAnonymisation`     | `bool?` | Anonymises certain user identifiers.                                                                                                                                              | ✔ | ✔ | ✔ | false                                         |
+| `lifecycleAutotracking` | `bool?` | Indicates whether the [lifecycle](iglu:com.snowplowanalytics.mobile/application_lifecycle/jsonschema/1-0-0) entity and events should be autotracked.                              | ✔ | ✔ | | false                                         |
```
Comment (matus-tomlein) [inline doc/02-configuration.md:50]:
```suggestion
| `lifecycleAutotracking` | `bool?` | Indicates whether the [lifecycle](iglu:com.snowplowanalytics.mobile/application_lifecycle/jsonschema/1-0-0) entity and foreground and background events should be autotracked (only on iOS and Android). | ✔ | ✔ | | false |
```

File: doc/02-configuration.md [hunk 2]
```diff
@@ -34,23 +34,26 @@ Setting a custom POST path can be useful in avoiding adblockers; it replaces the
...
+Activity tracking monitors whether a user continues to engage with a page over time, and record how they digest content on the page over time.  That is accomplished using 'page ping' events. If activity tracking is enabled, the web page is monitored to see if a user is engaging with it. (E.g. is the tab in focus, does the mouse move over the page, does the user scroll etc.) If any of these things occur in a set period of time (`minimumVisitLength` seconds from page load and every `heartbeatDelay` seconds after that), a page ping event fires, and records the maximum scroll left / right and up / down in the last ping period. If there is no activity in the page (e.g. because the user is on a different tab in their browser), no page ping fires.
+
+Lifecycle autotracking is for mobile apps. When configured, a Lifecycle context entity is attached to all events. It records whether the app was visible or not when the event was tracked. In addition, a `Background` event will be tracked when the app is moved to background, and a `Foreground` event when the app moves back to foreground (becomes visible on the screen).
```
Comment (matus-tomlein) [inline doc/02-configuration.md:56]:
```suggestion
Lifecycle autotracking is only available on mobile apps (iOS and Android). When configured, a Lifecycle context entity is attached to all events. It records whether the app was visible or not when the event was tracked. In addition, a `Background` event will be tracked when the app is moved to background, and a `Foreground` event when the app moves back to foreground (becomes visible on the screen).
```

**Review by matus-tomlein — APPROVED** [review]
LGTM!

File: doc/02-configuration.md (hunk 1 above)
Comment (mscwilson) [reply doc/02-configuration.md]: iOS and Android are indicated already by the table columns so I didn't add this change

**Pull request merged**